source/TimePicker.js

```diff
@@ -585,14 +608,29 @@
 		meridiemPickerChanged: function (sender, event) {
 			if(this.syncingPickers) return true;
 
-			var hour = this.$.hour.get('value'),
-				value = event.value;
-
-			if (this.value) {
-				// value is 0 for am, 1 for pm
-				// reset the hour to < 12 and then add 12 if it's pm
-				hour = hour%12 + value*12;
-				this.updateHours(hour);
+			var meridiems = this.meridiems,
+				hour = this.$.hour.getValue(),
+				oldIndex = event.old,
+				newIndex = this.$.meridiem.getValue(),
+				offset, start, end, newHour;
+
+			/*jshint -W030 */
+			offset = hour - parseInt(meridiems[oldIndex]['start'], 10);
+			start = parseInt(meridiems[newIndex]['start'], 10);
+			end = parseInt(meridiems[newIndex]['end'], 10);
+			newHour = offset + start;
+			/*jshint +W030 */
+
+			if (newHour != hour) {
+				if (newHour > end) {
+					newHour = end;
+				}
+				this.updateValue('hour', this.value.getHours() + (newHour - hour));
+			} else {
+				// there is only single case. 
+				// Ethiopia calendar has different meridiem status between 06:00 and 06:01
```

Comment:
Yes! I leave meridiem info for Ethiopia for you.

{ name: 'ጠዋት',
    start: '00:00',
    end: '05:59' },
  { name: 'ቀትር',
    start: '06:00',
    end: '06:00' },
  { name: 'ከሰዓት',
    start: '06:01',
    end: '11:59' },
  { name: 'ከምሽቱ',
    start: '12:00',
    end: '17:59' },
  { name: 'ከሌሊቱ',
    start: '18:00',
    end: '23:59' } ]

That's so interesting :)